rcl_lifecycle/test/test_rcl_lifecycle.cpp

```diff
@@ -228,148 +239,159 @@ TEST(TestRclLifecycle, state_machine) {
-  EXPECT_EQ(rcl_lifecycle_state_machine_is_initialized(&state_machine), RCL_RET_ERROR);
+  EXPECT_EQ(RCL_RET_ERROR, rcl_lifecycle_state_machine_is_initialized(&state_machine));
   rcutils_reset_error();
 
   ret = rcl_lifecycle_state_machine_init(
     &state_machine, &node, pn, cs, gs, gas, gat, gtg, false, nullptr);
-  EXPECT_EQ(ret, RCL_RET_ERROR);
+  EXPECT_EQ(RCL_RET_ERROR, ret);
   rcutils_reset_error();
-  EXPECT_EQ(rcl_lifecycle_state_machine_is_initialized(&state_machine), RCL_RET_ERROR);
+  EXPECT_EQ(RCL_RET_ERROR, rcl_lifecycle_state_machine_is_initialized(&state_machine));
   rcutils_reset_error();
 
 
   // Everything should be good
   ret = rcl_lifecycle_state_machine_init(
     &state_machine, &node, pn, cs, gs, gas, gat, gtg, false, &allocator);
-  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 
   // Transition_map is not initialized
-  EXPECT_EQ(rcl_lifecycle_state_machine_is_initialized(&state_machine), RCL_RET_ERROR);
+  EXPECT_EQ(RCL_RET_ERROR, rcl_lifecycle_state_machine_is_initialized(&state_machine));
   rcutils_reset_error();
 
-  void * temp_function = state_machine.com_interface.srv_change_state.impl;
-  state_machine.com_interface.srv_change_state.impl = nullptr;
-  // get_state service is valid, but not change_state service
-  EXPECT_EQ(rcl_lifecycle_state_machine_is_initialized(&state_machine), RCL_RET_ERROR);
-  rcutils_reset_error();
-  state_machine.com_interface.srv_change_state.impl =
-    reinterpret_cast<rcl_service_impl_t *>(temp_function);
-
   // Allocate some memory and initialize states and transitions so is_initialized will pass
   state_machine.transition_map.states_size = 1u;
   state_machine.transition_map.states = reinterpret_cast<rcl_lifecycle_state_t *>(
     allocator.allocate(
       state_machine.transition_map.states_size * sizeof(rcl_lifecycle_state_t),
       allocator.state));
-  ASSERT_NE(state_machine.transition_map.states, nullptr);
+  ASSERT_NE(nullptr, state_machine.transition_map.states);
   state_machine.transition_map.states[0] = rcl_lifecycle_get_zero_initialized_state();
 
   state_machine.transition_map.transitions_size = 1u;
   state_machine.transition_map.transitions =
     reinterpret_cast<rcl_lifecycle_transition_t *>(allocator.allocate(
       state_machine.transition_map.transitions_size * sizeof(rcl_lifecycle_transition_t),
       allocator.state));
-  ASSERT_NE(state_machine.transition_map.transitions, nullptr);
+  ASSERT_NE(nullptr, state_machine.transition_map.transitions);
   state_machine.transition_map.transitions[0] = rcl_lifecycle_get_zero_initialized_transition();
 
-  EXPECT_EQ(rcl_lifecycle_state_machine_is_initialized(&state_machine), RCL_RET_OK);
+  EXPECT_EQ(RCL_RET_OK, rcl_lifecycle_state_machine_is_initialized(&state_machine));
+
+  // get_state service is valid, but not change_state service
+  void * temp_function = state_machine.com_interface.srv_change_state.impl;
```

Comment:
is this temporal a function?

Reply:
Correct, `srv_change_state.impl` is a function pointer.